ledger/store/src/helpers/rocksdb/internal/map.rs

```diff
@@ -90,6 +89,9 @@ impl<
     fn start_atomic(&self) {
         // Set the atomic batch flag to `true`.
         self.batch_in_progress.store(true, Ordering::SeqCst);
+        // Increment the atomic depth index.
+        self.database.atomic_depth.fetch_add(1, Ordering::SeqCst);
+
```

Comment:
We may want to check that the underlying `atomic_batch` in `self.database` is empty.

```suggestion
// Ensure that the database atomic batch is empty.
assert!(self.database.atomic_batch.lock().is_empty());
```